source/kraken/maintenance_worker.cpp

```diff
@@ -82,69 +85,110 @@ void MaintenanceWorker::operator()(){
+    auto data = boost::shared_ptr<nt::Data>{nullptr};
+    for (auto& envelope: envelopes) {
+        LOG4CPLUS_DEBUG(logger, "realtime info received!");
+        assert(envelope);
+        transit_realtime::FeedMessage feed_message;
+        if(! feed_message.ParseFromString(envelope->Message()->Body())){
+            LOG4CPLUS_WARN(logger, "protobuf not valid!");
+            return;
+        }
+        LOG4CPLUS_TRACE(logger, "received entity: " << feed_message.DebugString());
+        for(const auto& entity: feed_message.entity()){
+            if (!data) {
+                data = data_manager.get_data_clone();
+                data->last_rt_data_loaded = pt::microsec_clock::universal_time();
+            }
+            if (entity.is_deleted()) {
+                LOG4CPLUS_DEBUG(logger, "deletion of disruption " << entity.id());
+                delete_disruption(entity.id(), *data->pt_data, *data->meta);
+            } else if(entity.HasExtension(chaos::disruption)) {
+                LOG4CPLUS_DEBUG(logger, "add/update of disruption " << entity.id());
+                add_disruption(entity.GetExtension(chaos::disruption), *data->pt_data, *data->meta);
+            } else if(entity.has_trip_update()) {
+                LOG4CPLUS_DEBUG(logger, "RT trip update" << entity.id());
+                handle_realtime(entity.trip_update(), *data);
+            } else {
+                LOG4CPLUS_WARN(logger, "unsupported gtfs rt feed");
+            }
         }
-        if (entity.is_deleted()) {
-            LOG4CPLUS_DEBUG(logger, "deletion of disruption " << entity.id());
-            delete_disruption(entity.id(), *data->pt_data, *data->meta);
-        } else if(entity.HasExtension(chaos::disruption)) {
-            LOG4CPLUS_DEBUG(logger, "add/update of disruption " << entity.id());
-            add_disruption(entity.GetExtension(chaos::disruption), *data->pt_data, *data->meta);
-        } else if(entity.has_trip_update()) {
-            LOG4CPLUS_DEBUG(logger, "RT trip update" << entity.id());
-            handle_realtime(entity.trip_update(), *data);
-        } else {
-            LOG4CPLUS_WARN(logger, "unsupported gtfs rt feed");
+    }
+    if (data) {
+        LOG4CPLUS_INFO(logger, "rebuilding data raptor");
+        data->build_raptor();
+        data_manager.set_data(std::move(data));
+        LOG4CPLUS_INFO(logger, "data updated");
+    }
+}
+
+std::vector<AmqpClient::Envelope::ptr_t>
+MaintenanceWorker::consume_in_batch(const std::string& consume_tag,
+        size_t max_nb,
+        size_t timeout_ms){
+    assert(consume_tag != "");
+    assert(max_nb);
+
+    bool not_empty = true;
+    std::vector<AmqpClient::Envelope::ptr_t> envelopes;
+    envelopes.reserve(max_nb);
+    size_t consumed_nb = 0;
+    while(not_empty && consumed_nb < max_nb) {
+        AmqpClient::Envelope::ptr_t envelope = nullptr;
+        not_empty = channel->BasicConsumeMessage(consume_tag, envelope, timeout_ms);
```

Comment:
wouldn't it be clearer to break on `! not_empty` (even if the double negation is not nice)